src/components/Search/SearchBox.tsx

```diff
@@ -0,0 +1,145 @@
+import { Button, Divider, Flex, Input, Text, VStack } from 'native-base';
+import { useEffect, useState } from 'react';
+import {
+  UserSearchResult,
+  buildUserSubstringMatcher,
+  getActiveRoutesCursor,
+  getArchivedRoutesSubstringMatcher,
+  getRouteByName,
+  getUserCache
+} from '../../xplat/api';
+import { User, Route, SubstringMatcher } from '../../xplat/types';
+import { UserRow } from '../UserRow';
+import { RouteRow } from '../RouteRow';
+import { useQuery } from 'react-query';
+
+export const enum SearchView {
+  Users,
+  ArchivedRoutes,
+  ActiveRoutes,
+}
+
+type Props = {
+  view: SearchView;
+  onSelect: (docRefID: string) => void;
+};
+export const SearchBox = ({ view, onSelect }: Props) => {
+  const [inputText, setInputText] = useState<string>('');
+  const [results, setResults] = useState<JSX.Element[]>([]);
+
+  const userMatcherQuery = useQuery(
+    [SearchView.Users, 'matcher'],
+    async () => buildUserSubstringMatcher(await getUserCache()),
+    {
+      enabled: view === SearchView.Users,
+    }
+  );
+
+  const activeRoutesQuery = useQuery(
+    'activeRoutes',
+    async () => getActiveRoutesCursor().________getAll_CLOWNTOWN_LOTS_OF_READS(),
+    {
+      enabled: view === SearchView.ActiveRoutes,
+    }
+  );
+  const activeRouteMatcherQuery = useQuery(
+    [SearchView.ActiveRoutes, 'matcher'],
+    async () => {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const routeNames: string[] = await Promise.all(activeRoutesQuery.data!.map((route: Route) => route.getName()));
+      return new SubstringMatcher<string>(routeNames);
+    },
+    {
+      enabled: view === SearchView.ActiveRoutes &&
+        !activeRoutesQuery.isLoading && !activeRoutesQuery.isError && activeRoutesQuery.data !== undefined,
+    }
+  );
+
+  const archivedRouteQuery = useQuery(
+    [SearchView.ArchivedRoutes, 'matcher'],
+    async () => getArchivedRoutesSubstringMatcher(),
+    {
+      enabled: view === SearchView.ArchivedRoutes,
+    }
+  );
+
+  // reset results for when our view prop changes
+  useEffect(() => {
+    setResults([]);
+  }, [view]);
+
+  // get which matcher query we are referring to for checking for loading and errors
+  const matcherQuery = (view === SearchView.Users ? userMatcherQuery :
```

Comment:
you should put this inside of the useEffect since this logic should only run when view changes

Reply:
so i am only using this variable to check if our useQuery is still loading or not, i end up using the actual 3 different queries when updating search results. Basically, this variable is a temp placeholder for the next two if statements. Should I still make it a piece of state?